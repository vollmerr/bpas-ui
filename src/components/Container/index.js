import React from 'react';
import PropTypes from 'prop-types';
import Grid from 'react-bootstrap/es/Grid';

function Container({
  children,
  ...props,
}){
  return <Grid {...props}>{children}</Grid>;
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Container;
