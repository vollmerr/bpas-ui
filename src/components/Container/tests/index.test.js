import React from 'react';
import { shallow } from 'enzyme';
import Container from '../';

describe('<Container />', () => {
  const Child = () => <div>child component</div>;
  const testClass = 'test-class';

  it('renders correctly', () => {
    const tree = shallow(
      <Container>
        <Child />
      </Container>
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders its children', () => {
    const wrapper = shallow((
      <Container>
        <Child />
      </Container>
    ));
    expect(wrapper.contains(<Child />)).toEqual(true);
  });

  it('accepts addditional props', () => {
    const wrapper = shallow(
      <Container className={testClass}>
        <Child />
      </Container>
    );
    expect(wrapper.props()).toHaveProperty('className', testClass);
  });
});
