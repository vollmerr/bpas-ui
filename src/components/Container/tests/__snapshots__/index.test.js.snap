// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> renders correctly 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
>
  <Child />
</Grid>
`;
